lib/spack/llnl/util/cpu/detect.py

```diff
@@ -109,20 +109,64 @@ def sysctl(*args):
         'model name': sysctl('-n', 'machdep.cpu.brand_string')
     }
 
-    # Super hacky way to deal with slight representation differences
-    # Would be better to somehow consider these "identical"
-    if 'sse4.1' in info['flags']:
+    adjust_raw_flags(info)
+
+    return info
+
+
+def adjust_raw_flags(info):
+    """Adjust the flags detected on the system to homogenize
+    slightly different representations.
+    """
+    # Flags detected on Darwin turned to their linux counterpart
+    flags = info.get('flags', [])
+    if 'sse4.1' in flags:
         info['flags'] += ' sse4_1'
-    if 'sse4.2' in info['flags']:
+    if 'sse4.2' in flags:
         info['flags'] += ' sse4_2'
-    if 'avx1.0' in info['flags']:
+    if 'avx1.0' in flags:
         info['flags'] += ' avx'
-    if 'clfsopt' in info['flags']:
+    if 'clfsopt' in flags:
         info['flags'] += ' clflushopt'
-    if 'xsave' in info['flags']:
+    if 'xsave' in flags:
         info['flags'] += ' xsavec xsaveopt'
```

Comment:
Probably the flag mappings above should go in the JSON as well (see my other comment)

Reply:
Done